Modern/wrapperPlugins/src/main/java/org/bonej/wrapperPlugins/wrapperUtils/UsageReporterOptions.java

```diff
@@ -0,0 +1,135 @@
+
+import org.scijava.ItemVisibility;
+import org.scijava.command.Command;
+import org.scijava.command.ContextCommand;
+import org.scijava.plugin.Parameter;
+import org.scijava.plugin.Plugin;
+import org.scijava.widget.Button;
+
+/**
+ * Handles persistent settings such as user permission state.
+ * 
+ * Preferences are stored in their native format (long, int, boolean, etc.)
+ * 
+ * @author Michael Doube
+ * @author Richard Domander
+ *
+ */
+@Plugin(type = Command.class, menuPath = "Edit>Options>BoneJ Usage (Modern)")
+public class UsageReporterOptions extends ContextCommand {
+
+	@Parameter(visibility = ItemVisibility.MESSAGE)
+	private String message1 = "Allow usage data collection?";
+	@Parameter(visibility = ItemVisibility.MESSAGE)
+	private String message2 = "BoneJ would like to collect data on";
+	@Parameter(visibility = ItemVisibility.MESSAGE)
+	private String message3 = "which plugins are being used, to direct development"; 
+	@Parameter(visibility = ItemVisibility.MESSAGE)
+	private String message4 = "and promote BoneJ to funders.";
+	@Parameter(visibility = ItemVisibility.MESSAGE)
+	private String message5 = "If you agree to participate please check the box.";		
+	@Parameter(visibility = ItemVisibility.MESSAGE)
+	private String helpMessage = "For more information click Help.";
+	
+	@Parameter(label = "Opt in to usage data collection", description = "Can BoneJ send usage data?")
+  private boolean optIn = false;
+	
+	@Parameter(label = "Help")
+	private Button button;
+	
+	@Parameter
+	private ImageJ imagej;
+
+	/** set to true if user permission has been requested */
+	static final String OPTINSET = "bonej2.report.option.set";
+	/** Set to false if reporting is not allowed */
+	static final String OPTINKEY = "bonej2.allow.reporter";
+	static final String COOKIE = "bonej2.report.cookie";
+  static final String COOKIE2 = "bonej2.report.cookie2";
+  /** time of first visit in seconds */
+	static final String FIRSTTIMEKEY = "bonej2.report.firstvisit";
+	/** unique ID for this particular BoneJ session */
+	static final String SESSIONKEY = "bonej2.report.bonejsession";
+	static final String IJSESSIONKEY = "bonej2.report.ijsession";
+
+	@Override
+	public void run() {
+
+		//TODO link to detailed online info
+    //For more information click Help.";
+		
+		//Wipe persistent data on opt-out
+		if (!optIn) {
+			System.out.print("User has opted out of data collection\n");
+			imagej.prefs().clear(this.getClass());
+		}
+
+		else {
+			System.out.print("User has opted in to data collection\n");
+			imagej.prefs().put(this.getClass(), OPTINKEY, true);
+			imagej.prefs().put(this.getClass(), COOKIE,
+				new Random().nextInt(Integer.MAX_VALUE));
+			imagej.prefs().put(this.getClass(), COOKIE2,
+				new Random().nextInt(Integer.MAX_VALUE));
+			imagej.prefs().put(this.getClass(), FIRSTTIMEKEY,
+				System.currentTimeMillis() / 1000);
+			imagej.prefs().put(this.getClass(), SESSIONKEY, 1);
+			imagej.prefs().put(this.getClass(), IJSESSIONKEY, 1);
```

Comment:
`this` qualifiers are redundant

Reply:
Could be a helper method `writePreferences`

Reply:
> `this` qualifiers are redundant

Fixed in f2ad3efe